src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -0,0 +1,459 @@
+import static org.folio.rest.support.http.InterfaceUrls.locCampusStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.locInstitutionStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.locLibraryStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.locationsStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.materialTypesStorageUrl;
+import static org.folio.util.StringUtil.urlEncode;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.text.IsEmptyString.isEmptyString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.rest.jaxrs.model.Items;
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+import org.folio.rest.support.client.LoanTypesClient;
+import org.folio.rest.support.client.MaterialTypesClient;
+import org.folio.rest.support.http.InterfaceUrls;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+
+/**
+ * Test cases to verify effectiveLocationId property calculation that implemented
+ * as two triggers for holdings_record and item tables (see itemEffectiveLocation.sql)
+ */
+@RunWith(JUnitParamsRunner.class)
+public class ItemEffectiveLocationTest extends TestBaseWithInventoryUtil {
+  private static final String PERMANENT_LOCATION_ID_KEY = "permanentLocationId";
+  private static final String TEMPORARY_LOCATION_ID_KEY = "temporaryLocationId";
+  private static final String EFFECTIVE_LOCATION_ID_KEY = "effectiveLocationId";
+
+  private static final String MAIN_LIBRARY_LOCATION = "Main Library (Item)";
+  private static final String SECOND_FLOOR_LOCATION = "Second Floor (item)";
+  private static final String ANNEX_LIBRARY_LOCATION = "Annex Library (item)";
+  private static final String ONLINE_LOCATION = "Online (item)";
+
+  private static String journalMaterialTypeID;
+  private static String canCirculateLoanTypeID;
+  private static UUID instanceId;
+  private static UUID mainLibraryLocationId;
+  private static UUID annexLibraryLocationId;
+  private static UUID onlineLocationId;
+  private static UUID secondFloorLocationId;
+
+  private static UUID loc1 = UUID.fromString("11111111-1111-4111-8111-111111111111");
```

Comment:
@julianladisch, why we can not use the locations above but creating this dummies instead?

Reply:
These are not dummies.
`beforeAny()` will create fully-fledged locations using that location id.
`PermTemp.toString()` uses the first digit of the uuid therefore I use a predefined value and not a random like the other ones. These predefined numbers improve readability in `parameters()` and in IDE's JUnit view.
Note that JUnit runs `parameters()` *before* `beforeAny()`.

Reply:
@julianladisch, I actually meant why we have separate locations for those two parametrized tests and can not reuse existing locations. I've already pushed some refactoring, so no prob on this.